spec/domain/monitor/messages_spec.rb

```diff
@@ -0,0 +1,35 @@
+RSpec.describe Monitor::Messages do
+  before do
+    allow(GovukStatsd).to receive(:increment)
+  end
+
+  it 'increments major if routing key ends .major' do
+    expect(GovukStatsd).to receive(:increment).with("major")
+    Monitor::Messages.run('news_story.major')
+  end
+
+  it 'increments minor if routing key ends .minor' do
+    expect(GovukStatsd).to receive(:increment).with("minor")
+    Monitor::Messages.run('news_story.minor')
+  end
+
+  it 'increments links if routing key ends .links' do
+    expect(GovukStatsd).to receive(:increment).with("links")
+    Monitor::Messages.run('news_story.links')
+  end
+
+  it 'increments republish if routing key ends .republish' do
+    expect(GovukStatsd).to receive(:increment).with("republish")
+    Monitor::Messages.run('news_story.republish')
+  end
+
+  it 'increments unpublish if routing key ends .unpublish' do
+    expect(GovukStatsd).to receive(:increment).with("unpublish")
+    Monitor::Messages.run('news_story.unpublish')
+  end
+
+  it 'increments discarded if routing key ends .discarded' do
+    expect(GovukStatsd).to receive(:increment).with("discarded")
+    Monitor::Messages.run('news_story.discarded')
```

Comment:
`news_story.discarded` will never be triggered. This is something to be handled separately by the Job that processes the messages.